### PR TITLE
Enabled aggregate API for nested controller

### LINKED ISSUE #166: OLM need work on aggregation API but it's disabled by Virtual Cluster
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]

OLM needs to work on Aggregation API for extension apiserver `packageserver`, but it's disabled by Virtual Cluster.
Please enable the aggregation layer via the following kube-apiserver flags, follows a example:
 ```
        - --enable-aggregator-routing=true
        - --requestheader-client-ca-file=/etc/kubernetes/pki/root/tls.crt
        - --requestheader-allowed-names=""
        - --requestheader-username-headers=X-Remote-User
        - --requestheader-group-headers=X-Remote-Group
        - --requestheader-extra-headers-prefix=X-Remote-Extra-
        - --proxy-client-key-file=/etc/kubernetes/pki/apiserver/tls.key
        - --proxy-client-cert-file=/etc/kubernetes/pki/apiserver/tls.crt
```

Otherwise，
```console
# oc get apiservice v1.packages.operators.coreos.com
NAME                                          SERVICE                     AVAILABLE                      AGE
v1.packages.operators.coreos.com              olm/packageserver-service   False (FailedDiscoveryCheck)   3h38m

# oc get csv packageserver -n olm
NAME            DISPLAY          VERSION   REPLACES   PHASE
packageserver   Package Server   0.18.1               Installing
```
Please reference more from: https://kubernetes.io/docs/tasks/extend-kubernetes/configure-aggregation-layer/

**What did you expect to happen:**
The OLM can be installed successfully.
```console
# oc get apiservice v1.packages.operators.coreos.com
NAME                               SERVICE                     AVAILABLE   AGE
v1.packages.operators.coreos.com   olm/packageserver-service   True        7h33m

# oc get csv packageserver -n olm
NAME            DISPLAY          VERSION   REPLACES   PHASE
packageserver   Package Server   0.18.1               Succeeded
```

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

I validated the solution by the following workaround:
`oc edit sts apiserver -n default-394ccf-vc-sample-1`
To add the following flags in args:
```
        - --enable-aggregator-routing=true
        - --requestheader-client-ca-file=/etc/kubernetes/pki/root/tls.crt
        - --requestheader-allowed-names=""
        - --requestheader-username-headers=X-Remote-User
        - --requestheader-group-headers=X-Remote-Group
        - --requestheader-extra-headers-prefix=X-Remote-Extra-
        - --proxy-client-key-file=/etc/kubernetes/pki/apiserver/tls.key
        - --proxy-client-cert-file=/etc/kubernetes/pki/apiserver/tls.crt

```
Then,  delete to restart apiserver-0 in default-394ccf-vc-sample-1 for enabling Aggregation API feature.

**Environment:**

- cluster-api-provider-nested version: built from master branch
- Minikube/KIND version: OpenShift 4.7.13
- Kubernetes version: (use `kubectl version`): 1.20.0
- OS (e.g. from `/etc/os-release`):  Red Hat Enterprise Linux 8

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api-provider-nested/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #166 

This is a follow up for https://github.com/kubernetes-sigs/cluster-api-provider-nested/issues/166 to fix the issue in nested controller.

/cc @chrishenzie @Fei-Guo

### CODE REVIEW

**Comment by chrishenzie** [comment]
/uncc

I lack context on this part of the codebase. Did you mean to CC @christopherhein ? (included in the linked PR #166)

File: controlplane/nested/certificate/consts.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ const (
 	// KubeletClient defines the client cert name for kubelet.
 	KubeletClient secret.Purpose = "kubelet-client"
 
-	// ProxyClient defines the client cert name for the front proxy.
-	ProxyClient secret.Purpose = "proxy-client"
+	// FrontProxyClient defines the client cert name for the front proxy.
```
Comment (christopherhein) [inline controlplane/nested/certificate/consts.go:39]: Can we leave this as `proxy` this follows the base standards from CAPI see - https://cluster-api.sigs.k8s.io/tasks/certs/using-custom-certificates.html#using-custom-certificates

**Review by christopherhein — CHANGES_REQUESTED** [review]
nit. switch back to `Proxy` for CAPI consistency.

File: controlplane/nested/certificate/consts.go (hunk 1 above)
Comment (gyliu513) [reply controlplane/nested/certificate/consts.go:39]: Done

**Review by christopherhein — APPROVED** [review]
awesome job! thank you!

/lgtm
/approve

**Pull request merged**